text/0041-Recoverable-fee-history-for-FOG-compatible-gift-codes.md

```diff
@@ -0,0 +1,137 @@
+stage of the gift code flow. A full 8 bytes could be used, but fees are not likely to be within the range of a 64 bit 
+number and it is desired to save space for descriptive utf-8 bytes.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The following proposed changes to each memo type are described below.
+
+## 0x0201 Gift code funding memo
+
+### Current description
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 4      |  First 4 bytes of blake2b over `tx_out_public_key` of gift code tx out |
+| 4 - 64     | "Note": A UTF-8 string with that is either exactly 60 bytes or null terminated if under 60 bytes |
+
+### Proposed change
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 4      |  First 4 bytes of blake2b over `tx_out_public_key` of gift code tx out |
+| 4 - 11      | Big-endian bytes of fee amount, as an unsigned 56-bit number |
+| 11 - 64     | "Note": A UTF-8 string with that is either exactly 60 bytes or null terminated if under 60 bytes |
+
+## 0x0202 Gift code cancellation memo
+
+### Current description
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 8      | `global_index` of gift code tx out that was cancelled |
+| 8 - 64     | Unused bytes |
+
+### Proposed change
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 8      | `global_index` of gift code tx out that was cancelled |
+| 8 - 15      | Big-endian bytes of fee amount, as an unsigned 56-bit number |
+| 15 - 64     | Unused bytes |
+
+## 0x0002 Gift code sender memo
+
+### Current description
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 64     | "Note": A UTF-8 string with that is either exactly 64 bytes or null terminated if under 64 bytes |
+
+### Proposed change
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 7      | Big-endian bytes of fee amount, as an unsigned 56-bit number |
+| 7 - 64     | "Note": A UTF-8 string with that is either exactly 64 bytes or null terminated if under 64 bytes |
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+By adding recoverable fee history, we take 7 bytes away from the utf-8 fields in the Funding and Sender memos which could
+be used to describe useful contextual information within the apps using them.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The rationale behind this is to create consistent accounting.
+
+However it could potentially be said that this is not strictly necessary and that it should be left up to implementing 
+apps to decide what to do with the utf-8 bytes. In a regime limited to 64 bytes or less, 7 bytes could be useful. 
+Therefore alternatives could be not to implement fee tracking at all OR to only use 6 bytes to track the fee.
+
+# Prior art
+[prior-art]: #prior-art
+
+[MCIP #32 - FOG-compatible-gift-codes](https://github.com/mobilecoinfoundation/mcips/blob/main/text/0032-fog-compatible-gift-codes.md)
```

Comment:
I think the original RTH MCIP that chose 7 bytes for the fees is also part of the prior art.